code/gradient.py

```diff
@@ -166,13 +166,14 @@ def _calculate_gradient_loss(self, d):
         return d_loss
 
     def _initialize(self):
+        import copy
         _X_std = self._X.std(axis=0) * 1/9
         _X_std[self._idv] = 0
         if self._idc is not None:
             for c in self._idc:
                 _X_std[c] = 0        
         _theta = np.random.normal(loc=0, scale=_X_std, size=self._X.shape)
-        self._X_changed = self._X + _theta
```

Comment:
Chciałem sprawdzić po co to jest i nie zauważyłem żadnej różnicy bez tego

Reply:
xD